### ISSUE TITLE
[Book Reports] htmlproofer claims URLs are not there

### ISSUE THREAD

**Issue by mtlynch** [mtlynch/mtlynch.io#332]
I'm getting odd errors from htmlproofer when I [try to add the Book Reports section](https://github.com/mtlynch/mtlynch.io/pull/329):

```
- ./_site/book-reports/fooled-by-randomness.html
  *  External link https://mtlynch.io/book-reports/fooled-by-randomness failed: 404 No error
- ./_site/book-reports/happy-city.html
  *  External link https://mtlynch.io/book-reports/happy-city failed: 404 No error
- ./_site/book-reports/index.html
  *  internally linking to /book-reports/fooled-by-randomness, which does not exist (line 410)
     <a href="/book-reports/fooled-by-randomness">
  <img src="/images/book-reports/fooled-by-randomness.jpg" alt="Fooled by Randomness by Nassim Nicholas Taleb" srcset="            https://mtlynch.io/images/resized/book-reports/320/fooled-by-randomness.jpg 320w,         https://mtlynch.io/images/book-reports/fooled-by-randomness.jpg 324w" sizes="175px" class="align-left" style="max-width:175px">
</a>
  *  internally linking to /book-reports/fooled-by-randomness, which does not exist (line 414)
     <a href="/book-reports/fooled-by-randomness"><em>Fooled by Randomness</em> by Nassim Nicholas Taleb</a>
  *  internally linking to /book-reports/fooled-by-randomness, which does not exist (line 419)
     <a href="/book-reports/fooled-by-randomness">Full notes</a>
  *  internally linking to /book-reports/happy-city, which does not exist (line 390)
     <a href="/book-reports/happy-city">
  <img src="/images/book-reports/happy-city.jpg" alt="Happy City by Charles Montgomery" srcset="            https://mtlynch.io/images/resized/book-reports/320/happy-city.jpg 320w,         https://mtlynch.io/images/book-reports/happy-city.jpg 327w" sizes="175px" class="align-left" style="max-width:175px">
</a>
  *  internally linking to /book-reports/happy-city, which does not exist (line 394)
     <a href="/book-reports/happy-city"><em>Happy City</em> by Charles Montgomery</a>
  *  internally linking to /book-reports/happy-city, which does not exist (line 399)
     <a href="/book-reports/happy-city">Full notes</a>
- ./_site/page5/index.html
  *  External link https://mtlynch.io/book-reports/ failed: 404 No error
```

Any idea what's going on here?

**Comment by mtlynch** [comment]
>For the "External link" errors, they are failing because the theme uses absolute URLs in certain areas so, as you can see it is trying to go out to https://mtlynch.io/book-reports but that of course hasn't been deployed yet, so therefore HTMLProofer is throwing errors. It looks like in #331 the dev build was removed which prevented HTMLProofer from testing absolute URLs to the production site, so that's why those are showing up now. I think we can still simply build the production site and use the url-swap option to change the absolute URLs to relative and that should solve those problems. I can test & PR this.

Ahhh, okay. Maybe that's why we set it up to build the dev site in the first place. I couldn't remember why we were doing that and it seemed like it might have just been accidental. If we can get it working by URL-swapping the absolute URLs, that sounds good. Otherwise, you can just submit a PR that's a revert of #331.

>However, I want to check first if /book-reports/happy-city.html is in fact the desired URL that you would like?

Oh, I want the permalink to be `/book-reports/happy-city` (no extension). Is that possible?

**Comment by mtlynch** [comment]
Closing as fixed in #333.
